alchemy/src/cloudflare/miniflare/remote-binding-proxy.ts

```diff
@@ -171,3 +194,61 @@ async function createWorkersPreviewSession(api: CloudflareApi) {
   const json: WorkersPreviewSession = await res.json();
   return json;
 }
+
+/**
+ * If the given domain uses Cloudflare Access, fetches the access token for the domain.
+ * Otherwise, returns undefined.
+ * @see https://github.com/cloudflare/workers-sdk/blob/a5eb5134f73d3983655d325a4de71c6370c57faa/packages/wrangler/src/user/access.ts#L10
+ */
+const getAccessToken = memoize(async (hostname: string) => {
+  if (!(await domainUsesAccess(hostname))) {
+    return undefined;
+  }
+  const child = spawn("cloudflared", ["access", "login", hostname], {
+    stdio: "pipe",
+  });
+  let stdout: string;
+  try {
+    [stdout] = await Promise.all([
+      child.stdout
+        .toArray()
+        .then((chunks) => chunks.map((chunk) => chunk.toString()).join("")),
+      once(child, "close"), // throws an error if the command fails
+    ]);
+  } catch (error) {
+    throw new Error(
+      [
+        `The \`cloudflared\` CLI is not installed, but is required to access the domain "${hostname}".`,
+        `Please install it from https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/installation and run \`cloudflared access login ${hostname}\`.`,
+      ].join("\n"),
+      { cause: error },
+    );
+  }
+  const matches = stdout.match(/fetched your token:\n\n(.*)/m);
+  if (matches?.[1]) {
+    return matches[1];
+  }
+  const error = new Error(
+    `Failed to get access token for domain "${hostname}".`,
+  );
+  Object.assign(error, { stdout });
+  throw error;
+});
+
+/**
+ * Returns true if the domain uses Cloudflare Access.
+ */
+async function domainUsesAccess(hostname: string) {
+  try {
+    const response = await fetch(`https://${hostname}`, {
+      redirect: "manual",
+      signal: AbortSignal.timeout(1000),
```

Comment:
I don't fully understand why the timeout is necessary, but [Wrangler does it](https://github.com/cloudflare/workers-sdk/blob/a5eb5134f73d3983655d325a4de71c6370c57faa/packages/wrangler/src/user/access.ts#L23-L31), so I assume it's there for a reason?

Reply:
Maybe the worker can hang?

This is a pretty awful thing to have to do. Are we hitting the user's data plane just to do an access check?